daemon/endpoint.go

```diff
@@ -1051,3 +1056,27 @@ func (e *endpointManager) DeleteEndpoint(ep *endpoint.Endpoint) int {
 func (e *endpointManager) deleteEndpointQuiet(ep *endpoint.Endpoint, conf endpoint.DeleteConfig) []error {
 	return ep.Delete(e.d, e.d.ipam, e, conf)
 }
+
+// Check verifies that endpoints are alive and healthy.
+func (e *endpointManager) Check(ep *endpoint.Endpoint) error {
+	// This only lives in endpointManager for convenience, the real
+	// intended home would be likely be pkg/datapath somewhere that
+	// understands how to gauge endpoint health from a node perspective.
+
+	// Be extra careful, we're only looking for one specific type of error
+	// currently: That the link has gone missing. Ignore other error to
+	// ensure that the caller doesn't unintentionally tear down the
+	// Endpoint thinking that it no longer exists.
+	_, err := netlink.LinkByName(ep.InterfaceName())
+	if _, ok := err.(netlink.LinkNotFoundError); ok {
```

Comment:
`errors.Is(err, netlink.LinkNotFoundError)`? (this might have been introduced in recent golang versions so the 1.7 branch might not have it)

Reply:
Yeah I went searching for this in the tree and couldn't find any examples in this branch. There are a few tidyups like this and context handling that we can do better in newer branches.

Reply:
Apparently `netlink.LinkNotFoundError` is a struct type, not a singleton error, so this can't be converted to `errors.Is(...)`:

> ../pkg/endpoint/bpf.go:1503:14: type netlink.LinkNotFoundError is not an expression

So I didn't change this for the newer version.